File: files/300-399/331/CH-331.py
import pandas as pd
import numpy as np

path = "300-399/331/CH-331 Pattern Combinations.xlsx"
input = pd.read_excel(path, usecols="B:C", skiprows=1, nrows=8)
test = pd.read_excel(path, usecols="E", skiprows=1, nrows=2).iloc[:, 0].tolist()

p1_str = "".join(np.where(input.index % 2 == 0, input.iloc[:, 0], input.iloc[:, 1]).astype(str))
p2_str = "".join(np.where(input.index % 2 == 0, input.iloc[:, 1], input.iloc[:, 0]).astype(str))

result = [p1_str, p2_str]
print(result == test) # True

File: files/300-399/331/CH-331.R
library(tidyverse)
library(readxl)

path <- "300-399/331/CH-331 Pattern Combinations.xlsx"
input <- read_excel(path, range = "B2:C9")
test <- read_excel(path, range = "E2:E4") %>% pull()

result <- input %>%
  mutate(
    p1 = ifelse(row_number() %% 2 == 1, `Column 1`, `Column 2`),
    p2 = ifelse(row_number() %% 2 == 1, `Column 2`, `Column 1`)
  ) %>%
  select(p1, p2) %>%
  summarise(across(everything(), ~ paste(., collapse = ""))) %>%
  pivot_longer(everything(), values_to = "pattern") %>%
  pull(pattern)

all.equal(result, test)
# [1] TRUE
